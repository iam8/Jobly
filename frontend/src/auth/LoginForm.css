.LoginForm {
    height: calc(100vh - 50px);
    display: flex;
    flex-direction: column;
    align-items: center;
}

.LoginForm-heading {
    padding-top: 3rem;
    padding-bottom: 1rem;
    color: white;
    text-shadow: 2px 2px 3px black;
}

.LoginForm-card {
    width: 30%;
}

.LoginForm label {
    font-weight: bold;
}
