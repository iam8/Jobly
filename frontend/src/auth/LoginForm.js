import React, {useState} from "react";
import { useHistory } from "react-router-dom";
import { Alert, Button, Card, CardBody, Form, FormGroup, Label, Input } from "reactstrap";

import "./LoginForm.css";


/**
 * Route: /login
 *
 * Form used to log in to Jobly website.
 *
 * Manages state updates on changes to the form inputs.
 *
 * On successful login, calls login (function prop) and redirects to / (home).
 *
 * On login failure, displays alert and error message.
 */
function LoginForm({login}) {
    const INIT_FORM = {
        username: "",
        password: ""
    };

    const [formData, setFormData] = useState(INIT_FORM);
    const [formErrors, setFormErrors] = useState([]);
    const history = useHistory();

    /** Handle changes to inputs. */
    const handleChange = (evt) => {
        const {name, value} = evt.target;
        setFormErrors([]);
        setFormData((formData) => ({
            ...formData,
            [name]: value
        }));
    }

    /** Form submission - log in user and redirect to home on success. */
    const handleSubmit = async (evt) => {
        evt.preventDefault();
        const result = await login(formData);

        if (result.success) {
            history.push("/"); // Redirect to homepage
        } else {
            setFormErrors(result.err);
        }
    }

    /** Show an alert message on failure to log in. */
    const renderAlert = () => {
        if (formErrors.length) {
            return (
                <Alert color="danger">
                    Login failed - {formErrors}
                </Alert>
            );
        }
    }

    return (
        <div className="LoginForm">
            <h2 className="LoginForm-heading">Log In</h2>

            <Card className="LoginForm-card">
                <CardBody>
                    <Form className="LoginForm-form" onSubmit={handleSubmit}>
                        <FormGroup>
                            <Label htmlFor="username">Username</Label>
                            <Input
                                id="username"
                                name="username"
                                type="text"
                                value={formData.username}
                                onChange={handleChange}
                            />
                        </FormGroup>

                        <FormGroup>
                            <Label htmlFor="password">Password</Label>
                            <Input
                                id="password"
                                name="password"
                                type="password"
                                value={formData.password}
                                onChange={handleChange}
                            />
                        </FormGroup>

                        {renderAlert()}

                        <Button color="primary" block>Submit</Button>
                    </Form>
                </CardBody>
            </Card>
        </div>
    );
}


export default LoginForm;
