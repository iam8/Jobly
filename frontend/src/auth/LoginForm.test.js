import React from "react";
import { render } from "@testing-library/react";
import LoginForm from "./LoginForm";


test("Renders without crashing", () => {
    render(<LoginForm />);
})


test("Matches snapshot", () => {
    const { asFragment } = render(
        <LoginForm />
    );

    expect(asFragment()).toMatchSnapshot();
})
