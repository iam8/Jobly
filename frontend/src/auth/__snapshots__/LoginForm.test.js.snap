// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoginForm"
  >
    <h2
      class="LoginForm-heading"
    >
      Log In
    </h2>
    <div
      class="LoginForm-card card"
    >
      <div
        class="card-body"
      >
        <form
          class="LoginForm-form"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="username"
            >
              Username
            </label>
            <input
              class="form-control"
              id="username"
              name="username"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="password"
            >
              Password
            </label>
            <input
              class="form-control"
              id="password"
              name="password"
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-primary d-block w-100"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
